lib/os/async.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "toolchain/gcc.h"
+#include <zephyr/sys/async.h>
+#include <zephyr/kernel.h>
+
+void async_semaphore_cb(void *context, int result, void *sem_data)
+{
+	ARG_UNUSED(context);
+
+	struct async_sem *a = (struct async_sem *)sem_data;
+
+	a->result = result;
+
+	k_sem_give(a->sem);
```

Comment:
Note: `context` here is not propagated to the thread blocking with `k_sem_take()`